### ISSUE TITLE
solcjs --standard-json - TypeError: Second argument needs to be a buffer

### ISSUE THREAD

**Issue by Janaka-Steph** [ethereum/solc-js#126]
Hello,

`solcjs --standard-json myinput.json` doesn't work. See the error bellow. 

```
$ echo "{}" | solcjs --standard-json
/usr/local/lib/node_modules/solc/node_modules/graceful-fs/polyfills.js:144
        throw er
        ^

TypeError: Second argument needs to be a buffer
    at Object.fs.readSync (fs.js:681:18)
    at Object.readSync (/usr/local/lib/node_modules/solc/node_modules/graceful-fs/polyfills.js:138:28)
    at Object.<anonymous> (/usr/local/lib/node_modules/solc/solcjs:65:18)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Function.Module.runMain (module.js:605:10)
    at startup (bootstrap_node.js:158:16)
```

**Comment by axic** [comment]
Thanks, this seems to be a bug. `solcjs --standardjson myinput.json` will not work as it expects the content on the standard input (ie. piping). `echo "{}" | solcjs --standard-json` should work though.

**Comment by axic** [comment]
Will push a fix soon.

**Comment by munawarb** [comment]
Hi,
I'm trying to execute this command: `echo "{}" | solcjs --standard-json` but getting an error: `Empty input was read`.

**Comment by chriseth** [comment]
@munawarb the empty object is not enough, some of the fields are required. More information: https://solidity.readthedocs.io/en/develop/using-the-compiler.html#compiler-input-and-output-json-description

**Comment by axic** [comment]
The `Empty input was read` is part of `solcjs` though, which is being addressed in #127.

**Comment by mtbitcoin** [comment]
I am running into the exact issue, is there a workaround or fix?

**Comment by munawarb** [comment]
Not that I know of. I just passed the source code as stdin. An example 
using Java:

```

         ProcessBuilder processBuilder = new ProcessBuilder("solc", 
"--combined-json", "abi,bin", "--optimize");
         processBuilder.redirectErrorStream(true);
         Process process = processBuilder.start();
         OutputStream stdin = process.getOutputStream();
         stdin.write(smartContractCode.getBytes());
         stdin.flush();
         stdin.close();

```

**Comment by axic** [comment]
@snd there is the PR #127 trying to fix this, but no matter what, something works on Travis but doesn't work on a Mac, or vice versa, but mostly nothing works on Travis. Would you be interested in helping out?

**Comment by tzapu** [comment]
@axic i tried to fix the tests here https://github.com/ethereum/solc-js/pull/229, would love to be a be able to use standard jsons :D
thanks

**Comment by feuGeneA** [comment]
@axic , what's your disposition on this issue and the proposed fix?

I'm using the changes from @tzapu 's PR #229 , which seem to include your own changes as well, and it's all working for my purposes.

I'd love to see the fix merged into master so that I can go into production with my own development that's depending on it, and I see @snd and others are waiting on it as well.

Are there any reservations you have?  Is there anything we can do to help facilitate?

**Comment by axic** [comment]
Here's the last comment on the issue: https://github.com/ethereum/solc-js/pull/127#discussion_r201709486
